src/nix-eval-jobs.cc

```diff
@@ -178,6 +178,9 @@ std::string joinAttrPath(json &attrPath) {
 }
 
 void collector(Sync<State> &state_, std::condition_variable &wakeup) {
+    struct GC_stack_base sb;
```

Comment:
This thread doesn't do any evaluation. The evaluation is running in a process.

Reply:
It could be that it still works because of the work...

Reply:
Correct. However Boehm needs to know about the thread that does the vfork. If it doesn't know about the thread doing the fork, then in the child process it will print the error displayed.

You can manually validate that the fix works by running `nix-eval-jobs` with the environment variable `GC_PRINT_STATS=1` (and removing the `GC_DONT_GC` environment variable). Before this change you'll see zero collections occur. After this change you'll see in the logs that collections do occur.